sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/src/LiveMetrics/Manager.Metrics.cs

```diff
@@ -132,6 +136,26 @@ public MonitoringDataPoint GetDataPoint()
             return dataPoint;
         }
 
+        private Request CreateUpdatedRequest(Request request, IList<string> documentStreamIds)
+        {
+            return new Request(request.DocumentType, documentStreamIds, request.Properties, request.Name, request.Url, request.ResponseCode, request.Duration);
+        }
```

Comment:
Can you find a way to add the documentStreamIds to the existing instance?

Reply:
We are creating the same object twice, this is a perf concern.

Reply:
Currently the field is readonly. This is the same issue as in the Error object's KeyValuePair list. Another approach on top of my mind is to make another class or field for mutable operations, and construct the readonly class instance or field by the time it's sent via network or passed onto other APIs.

Reply:
https://github.com/Azure/azure-sdk-for-net/blob/d800162c44ed315a8f6dc71f3eae9ebb3acabd26/sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/src/Generated/Models/DocumentIngress.cs#L40

You shouldn't need to overwrite the `DocumentStreamIds`
As this is a list, you can call the `Add` method.

```
item.DocumentStreamIds.Add(...)
```

Reply:
I've updated it.